jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/ReproducibleImageTest.java

```diff
@@ -0,0 +1,194 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.jib.api;
+
+import com.google.cloud.tools.jib.configuration.CacheDirectoryCreationException;
+import com.google.cloud.tools.jib.configuration.FilePermissions;
+import com.google.cloud.tools.jib.configuration.LayerConfiguration;
+import com.google.cloud.tools.jib.filesystem.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.image.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.registry.RegistryException;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.BiConsumer;
+import java.util.zip.GZIPInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * Verify that created image has explicit directory structures, default timestamps, permissions, and
+ * file orderings.
+ */
+public class ReproducibleImageTest {
+  private static String LAYERS_PATH_IN_RESOURCES = "layers/";
+
+  @ClassRule public static final TemporaryFolder imageLocation = new TemporaryFolder();
+
+  private static File imageTar;
+
+  @BeforeClass
+  public static void createImage()
+      throws InvalidImageReferenceException, InterruptedException, CacheDirectoryCreationException,
+          IOException, RegistryException, ExecutionException {
+
+    Path root = imageLocation.getRoot().toPath();
+    Path fileA = Files.createFile(root.resolve("fileA.txt"));
+    Path fileB = Files.createFile(root.resolve("fileB.txt"));
+    Path fileC = Files.createFile(root.resolve("fileC.txt"));
+    Path subdir = Files.createDirectory(root.resolve("dir"));
+    Path fileD = Files.createFile(subdir.resolve("fileD.txt"));
+
+    imageTar = new File(imageLocation.getRoot(), "image.tar");
+    Containerizer containerizer =
+        Containerizer.to(TarImage.named("jib-core/reproducible").saveTo(imageTar.toPath()));
+
+    ExecutorService executorService = Executors.newCachedThreadPool();
+    containerizer.setExecutorService(executorService);
```

Comment:
Do we have to set this? Jib by default will use `Executors.newCachedThreadPool()` and shut it down.

Reply:
Test copying.  You're right (I hope)